sdk/keyvault/azure-security-keyvault/azure/security/keyvault/secrets/README.md

```diff
@@ -0,0 +1,252 @@
+    from azure.security.keyvault.aio import SecretClient
+
+    credential = AsyncClientSecretCredential(client_id=client_id, secret=client_secret, tenant_id=tenant_id)
+    # Create a new secret client using a client secret credential
+    secret_client = SecretClient(vault_url=vault_url, credential=credential)
+
+    # Sends secret data and asynchronously waits until acknowledgement is received
+    secret = await secret_client.set_secret("secret-name", "secret-value", enabled=True)
+    
+    print(secret.id)
+    print(secret.value)
+    print(secret.version)
+    print(secret.enabled)
+```
+### Async list secrets
+This example lists all the secrets in the specified Key Vault.
+```python
+    import asyncio
+
+    from azure.identity import AsyncClientSecretCredential
+    from azure.security.keyvault.aio import SecretClient
+
+    credential = AsyncClientSecretCredential(client_id=client_id, secret=client_secret, tenant_id=tenant_id)
+    # Create a new secret client using a client secret credential
+    secret_client = SecretClient(vault_url=vault_url, credential=credential)
+
+    secrets = secret_client.list_secrets()
+
+    # Recieve secrets data asynchronously from Key Vault service
+    async for secret in secrets:
+        # the list doesn't include values or versions of the secrets
+        print(secret.id)
+        print(secret.name)
+```
+
+## Troubleshooting
+### General
+The Key Vault APIs generate exceptions that can fall into one of the azure-core defined exceptions. For more detailed infromation about exceptions and how to deal with them, see [Azure Core exceptions](TODO).
+
+For example, if you try to retrieve a secret after it is deleted a `404` error is returned, indicating resource not found. In the following snippet, the error is handled gracefully by catching the exception and displaying additional information about the error.
+```python
+try:
+    client.get_secret("deleted_secret")
+except ResourceNotFoundError as e:
+    if e.code == 404:
+        print(e.message)
+    else:
+        raise
+    
+Output: "Secret not found:deleted_secret"
+```
+### Logging [TODO]
+This SDK uses Python standard logging library. You can configure logging print out debugging information to the stdout or anywhere you want.
+
+```python 
+import logging
+logging.basicConfig(level=logging.DEBUG)
+```
+Http request and response details are printed to stdout with this logging config.
+
+## Next steps
+Several KeyVault Python SDK samples are available to you in the SDK's GitHub repository. These samples provide example code for additional scenarios commonly encountered while working with Key Vault:
+* [test_examples_keyvault.py](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/keyvault/azure-security-keyvault/tests/test_examples_keyvault.py) - Contains code snippets found in this article.
+* [test_examples_secrets_sync.py](TODO) -  Python code for working with Key Vault secrets, including:
```

Comment:
How is this one different from the one above?

Reply:
This is a TODO workitem, filenames would be changing. Will update this once I have that merged.

Reply:
Right, but I meant in terms of what the sample actually does (based on the description there) - the methods used in that code will be the same as the ones right here? Or will it show some other capabilities in action?

Reply:
yeah, at most there would be 2 file samples here.